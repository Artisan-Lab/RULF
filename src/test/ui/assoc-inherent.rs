// Test that inherent associated types work with
// inherent_associated_types feature gate.

#![feature(inherent_associated_types)]
#![allow(incomplete_features)]

struct Foo;

impl Foo {
    type Bar = isize;
}

impl Foo {
    type Baz; //~ ERROR associated type in `impl` without body
}

fn main() {
    let x : Foo::Bar; //~ERROR ambiguous associated type
    x = 0isize;
}
