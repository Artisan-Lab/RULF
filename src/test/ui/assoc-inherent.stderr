error: associated type in `impl` without body
  --> $DIR/assoc-inherent.rs:14:5
   |
LL |     type Baz;
   |     ^^^^^^^^-
   |             |
   |             help: provide a definition for the type: `= <type>;`

error[E0223]: ambiguous associated type
  --> $DIR/assoc-inherent.rs:18:13
   |
LL |     let x : Foo::Bar;
   |             ^^^^^^^^ help: use fully-qualified syntax: `<Foo as Trait>::Bar`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0223`.
